californium-core/src/main/java/org/eclipse/californium/core/coap/Message.java

```diff
@@ -438,6 +439,7 @@ public Message setToken(byte[] tokenBytes) {
 	public Message setToken(Token token) {
 		this.token = token;
 		bytes = null;
+		sent = false;
```

Comment:
Why setting Token or MID imply `sent = false` ?

Reply:
Because it resets also the bytes ...
The first commit uses `sent` as indicator for retransmission instead of initial tansmission. 
According the transmission layer, a message with a new MID is a new message and so no retransmission.
According the request/response layer, an new token is also a new request/response and so no retransmission.
Do you feel, that this causes some issues?
We may check, why these values are changed at all after serialization.

Reply:
Maybe it's better, not to support to alter the MID or token after serialization?
I did a first check, and that seems to work as well (with small changes).

Reply:
> Do you feel, that this causes some issues?

I don't know. My guess was Message is initialized with `sent = false`, then `sent` is set to `true` after once the message is sent.

So I was surprised that we need to set it to false on `setToken` or `setMID`.

> Maybe it's better, not to support to alter the MID or token after serialization?

If we could avoid it, this is probably better.

Reply:
Currently only the Matcher `sendResponse` was violating this.
I changed it, but I need to run the benchmark tests.
Maybe you check, if that setMID/Token after serialization is something leshan uses ...

Reply:
My tests have been successful.
Leshan unit tests also.
I squash and merge :-)

Reply:
setMID is never used in Leshan.
setToken only to create active cancel observe request. (done before we send the request)